### ISSUE TITLE
Systray position error on multiple monitor vertically aligned

### ISSUE THREAD

**Issue by herod2k** [polybar/polybar#1537]
## Describe the issue
The systray is in the wrong position when I use two monitors vertically aligned. It appears outside of Polybar

**Expected behavior:**
The systray should appear on the Polybar 

**Actual behavior:**
The systray appear on the second screen outside of Polybar

## To Reproduce
* use two monitors
* load some software with applet on systray (Telegram, Slack, udiskie, etc etc)
* Align the monitor vertically using arandr

## Polybar Log
```
error: module/xbacklight: Could not get data (err: XCB_NAME (15))
error: Disabling module "xbacklight" (reason: Not supported for "eDP-1")
error: tray: Failed to put tray above 0x1200002 in the stack (XCB_MATCH (8))
```

## Polybar Configuration
```
[colors]
color0 = ${xrdb:color0}
color1 = ${xrdb:color1}
color2 = ${xrdb:color2}
color3 = ${xrdb:color3}
color4 = ${xrdb:color4}
color5 = ${xrdb:color5}
color6 = ${xrdb:color6}
color7 = ${xrdb:color7}

background = ${xrdb:background}
foreground-alt = #555
foreground = ${xrdb:foreground}
primary = ${colors.color3}
alert = #bd2c40

[bar/b0]
width = 100%
height = 20
fixed-center = false
monitor = eDP-1

background = ${colors.background}
foreground = ${colors.foreground}

line-size = 3
line-color = ${colors.color1}

padding-left = 0
padding-right = 2

module-margin = 1

font-0 = FontAwesome:pixelsize=9;2
font-1 = xos4 Terminus:pixelsize=9;2
font-2 = Source\ Han\ Sans:pixelsize=9;2

modules-left = i3 pulseaudio previous playpause next spotify
modules-center = xwindow 
modules-right = filesystem xkeyboard cpu memory xbacklight battery date powermenu

locale = en_GB.utf8

tray-position = right
tray-padding = 2

enable-ipc = true

scroll-up = i3wm-wsnext
scroll-down = i3wm-wsprev

cursor-click = pointer

[module/xwindow]
type = internal/xwindow
label = %title:0:40:...%

[module/xkeyboard]
type = internal/xkeyboard
blacklist-0 = num lock

format-prefix = " "
format-prefix-foreground = ${colors.color1}

label-layout = %layout%

label-indicator-padding = 2
label-indicator-margin = 1
label-indicator-background = ${colors.alert}

[module/filesystem]
type = internal/fs
interval = 25

mount-0 = /

label-mounted = %{F#0a81f5}%mountpoint%%{F-}: %percentage_used%%
label-unmounted = %mountpoint% not mounted
label-unmounted-foreground = ${colors.foreground-alt}

[module/i3]
type = internal/i3
format = <label-state> <label-mode>
index-sort = true
wrapping-scroll = false

; Only show workspaces on the same output as the bar
pin-workspaces = true

label-mode-padding = 2
label-mode-foreground = ${colors.background}
label-mode-background = ${colors.primary}

; focused = Active workspace on focused monitor
label-focused = %name%
label-focused-underline= ${colors.primary}
label-focused-padding = 2

; unfocused = Inactive workspace on any monitor
label-unfocused = %name%
label-unfocused-padding = 2

; visible = Active workspace on unfocused monitor
label-visible = %name%
label-visible-underline = ${self.label-focused-underline}
label-visible-padding = ${self.label-focused-padding}

; urgent = Workspace with urgency hint set
label-urgent = %name%
label-urgent-background = ${colors.alert}
label-urgent-padding = 2

[module/mpd]
type = internal/mpd
format-online = <label-song>  <icon-prev> <icon-stop> <toggle> <icon-next>
format-stopped =

icon-prev = 
icon-stop = 
icon-play = 
icon-pause = 
icon-next = 

label-song-maxlen = 40
label-song-ellipsis = true

[module/xbacklight]
type = internal/xbacklight

format = <label> <bar>
label = 

bar-width = 10
bar-indicator = |
bar-indicator-foreground = #ff
bar-indicator-font = 2
bar-fill = ─
bar-fill-font = 2
bar-fill-foreground = #9f78e1
bar-empty = ─
bar-empty-font = 2
bar-empty-foreground = ${colors.foreground-alt}

[module/player-mpris-tail]
type = custom/script
exec = ~/.config/polybar/player-mpris-tail.py
tail = true
click-left = playerctl previous
click-right = playerctl next
click-middle = playerctl play-pause

[module/cpu]
type = internal/cpu
interval = 2
format-prefix = " "
format-prefix-foreground = ${colors.color2}
label = %percentage:2%%

[module/memory]
type = internal/memory
interval = 2
format-prefix = " "
format-prefix-foreground = ${colors.color3}
label = %percentage_used%%

[module/eth]
type = internal/network
interface = eno1
interval = 3.0

format-connected-underline = #55aa55
format-connected-prefix = " "
format-connected-prefix-foreground = ${colors.foreground}
label-connected = %local_ip%

format-disconnected =
;format-disconnected = <label-disconnected>
;format-disconnected-underline = ${self.format-connected-underline}
;label-disconnected = %ifname% disconnected
;label-disconnected-foreground = ${colors.foreground-alt}

[module/date]
type = internal/date
interval = 5

date = %a, %d %b %Y
date-alt= %Y-%m-%d
time = %H:%M
time-alt = %H:%M:%S

format-prefix = " "
format-prefix-foreground = ${colors.color4}

label = %date% - %time%

[module/pulseaudio]
type = internal/pulseaudio
sink = alsa_output.pci-0000_00_1f.3.analog-stereo
use-ui-max = true
interval = 5
format-volume = <ramp-volume> <label-volume>
label-muted = muted
label-muted-foreground = ${colors.foreground}
ramp-volume-0 =  
ramp-volume-1 = 
ramp-volume-2 = 

[module/battery]
type = internal/battery
battery = BAT0
adapter = AC
full-at = 99

format-charging = <animation-charging> <label-charging>

format-discharging = <ramp-capacity> <label-discharging>

format-full-prefix = " "
format-full-prefix-foreground = ${colors.color6}

ramp-capacity-0 = 
ramp-capacity-1 = 
ramp-capacity-2 = 
ramp-capacity-3 = 
ramp-capacity-4 = 
ramp-capacity-foreground = ${colors.color6}

animation-charging-0 = 
animation-charging-1 = 
animation-charging-2 = 
animation-charging-3 = 
animation-charging-4 = 
animation-charging-foreground = ${colors.color6}
animation-charging-framerate = 750

[module/powermenu]
type = custom/menu

expand-right = true

format-spacing = 1

label-open = 
label-open-foreground = ${colors.color1}
label-close =  cancel
label-close-foreground = ${colors.color1}
label-separator = |
label-separator-foreground = ${colors.foreground-alt}

menu-0-0 = log out
menu-0-0-exec = i3-msg exit
menu-0-1 = reboot
menu-0-1-exec = systemctl reboot
menu-0-2 = power off
menu-0-2-exec = systemctl poweroff

[module/playpause]
type = custom/script
exec = echo ""
format-padding = 1
format = <label>
format-foreground = ${colors.foreground}
format-background = ${colors.background}
click-left = playerctl play-pause

[module/next]
type = custom/script
exec = echo ""
format-padding = 1
format = <label>
format-foreground = ${colors.foreground}
format-background = ${colors.background}
click-left = playerctl next

[module/previous]
type = custom/script
exec = echo ""
format-padding = 1
format = <label>
format-foreground = ${colors.foreground}
format-background = ${colors.background}
click-left = playerctl previous


[module/spotify]
type = custom/script
exec = python ~/.config/polybar/spotify.py
interval = 5
format = <label>
label =%output:0:500:...%
format-background = ${colors.background}
format-padding = 3
format-foreground = ${colors.foreground}

[settings]
screenchange-reload = true
;compositing-background = xor
;compositing-background = screen
;compositing-foreground = source
;compositing-border = over

[global/wm]
margin-top = 5
margin-bottom = 5

; vim:ft=dosini
```

## Screenshots
Horizontal configuration, systray in correct position:
![2018-12-03-151441_3840x1080_scrot](https://user-images.githubusercontent.com/3713164/49383207-1a71c080-f710-11e8-86d2-fd831da82d29.png)
Vertical configuration, systray in wrong position:
![2018-12-03-151417_1920x2160_scrot](https://user-images.githubusercontent.com/3713164/49383213-1e054780-f710-11e8-8276-ca72323cf1d0.png)

## Environment:
Xubuntu 18.04 
`i3 version 4.16-37-gdbab1538 (2018-11-10, branch "gaps-next") © 2009 Michael Stapelberg and contributors`
```polybar 3.2.1

Features: -alsa +curl +i3 -mpd +network +pulseaudio +xkeyboard

X extensions: +randr (+monitors) -render -damage -sync -composite +xkb +xrm +xcursor

Build type: RelWithDebInfo
Compiler: /usr/bin/c++
Compiler flags:  -Wall -Wextra -Werror -Wno-noexcept-type -O2 -pedantic -pedantic-errors
Linker flags:
```

## Additional context
* As you can see I'm forcing the bar on my Laptop display, I'm not sure if it is relevant or not.

**Comment by herod2k** [comment]
Removing (and fixing) xbacklight from the configuration works. I didn't think that the two things were related. I didn't see the connection.

**Comment by patrick96** [comment]
Was the tray misbehaving consistently before? xbacklight shouldn't influence the tray at all, maybe it is something that doesn't always happen.

**Comment by herod2k** [comment]
Hi @patrick96 , yes it was a constant error. I've restarted it multiple time and it was always there in the vertical alignment. I've fixed the xbacklight error, I've restarted and now I don't have any problem. I've restarted and reloaded multiple time and it works perfectly. It is weird and I'm not sure that the errors are related. I've tried to reload xbacklight again, I get the xbacklight error but systry now it is working. So I don't know, I can't reproduce the error anymore but I don't think that I did much more than fixing the configuration.

**Comment by patrick96** [comment]
Maybe you also did some minor change to your screen setup. Report back, if you ever see the error again.

**Comment by herod2k** [comment]
Hi @patrick96 ,

I'm getting the error again. Same problem. (Today I have 3 monitors)

Correct version (with the main monitor on the left)
![2018-12-04-085422_5760x1080_scrot](https://user-images.githubusercontent.com/3713164/49430321-9c122e80-f7a2-11e8-8be4-623bc444658f.png)

Error with the monitors not aligned.
![2018-12-04-085533_3840x2160_scrot](https://user-images.githubusercontent.com/3713164/49430325-9f0d1f00-f7a2-11e8-9b49-aeaf8dcf44ba.png)

**Comment by patrick96** [comment]
Hmm, I'm also going to try and reproduce that, maybe I can find the bug.

This may be related to #1437.

In the meantime, can you recompile polybar while adding `-DWITH_XRANDR_MONITORS=OFF` to the `cmake` command. This seems to have helped in #1437

**Comment by herod2k** [comment]
Hi @patrick96 I've just tried but I don't see any changes. Same error.

**Comment by patrick96** [comment]
Hmm, then we'll need to reproduce this on our end to find the bug

**Comment by patrick96** [comment]
This is fixed by #2609 if you are using the new tray module.
